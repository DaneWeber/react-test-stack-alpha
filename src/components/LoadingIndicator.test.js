import React from "react";
import TestRenderer from "react-test-renderer";
import LoadingIndicator from "./LoadingIndicator";

describe("LoadingIndicator", () => {
  describe("when isLoading is false", () => {
    it("should render children", () => {
      // Arrange
      // Act
      const wrapper = TestRenderer.create(
        <LoadingIndicator isLoading={false}>
          <div>ahoy!</div>
        </LoadingIndicator>
      );

      // Assert
      const result = wrapper.root;
      expect(result.findByType("div").children[0]).toEqual("ahoy!");

      // The following doesn't work, while there is no visual difference.
      // expect(wrapper.toJSON()).toEqual(<div>ahoy!</div>);

      // Clean up
      wrapper.unmount();
    });
  });
});
