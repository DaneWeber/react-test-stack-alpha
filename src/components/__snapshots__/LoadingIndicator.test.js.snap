// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingIndicator when isLoading is false should match the snapshot 1`] = `
<div>
  ahoy!
</div>
`;
